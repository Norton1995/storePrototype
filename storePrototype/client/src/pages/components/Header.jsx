import React, { Component } from "react";
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink, } from 'reactstrap';

//Header component consists of a navbar that loads the varies routes declared in main App
class Header extends Component {
    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
        this.state = {
            isOpen: false
        };
    }
    toggle() {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }
    render() {
        return (
          <div>
            <div>
                <Navbar color="danger" dark expand="md">
                    <NavbarBrand href="/">Golden Shoe Store Prototype</NavbarBrand>
                    <NavbarToggler onClick={this.toggle} />
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                          <NavItem>
                              <NavLink href="/">Home</NavLink>
                          </NavItem>
                            <NavItem>
                                <NavLink href="/products/">Example Product Page</NavLink>
                            </NavItem>
                            <NavItem>
                              <NavLink href="/about/">Help</NavLink>
                          </NavItem>
                        </Nav>
                    </Collapse>
                </Navbar>
              </div>
            </div>
          );
        }
      }

export default Header;
