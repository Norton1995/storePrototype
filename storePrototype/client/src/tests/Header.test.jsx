import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../pages/components/Header';
import {configure, shallow} from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";

configure({ adapter: new Adapter() });

describe('Home renders correctly', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Page should match snapshot', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });
});