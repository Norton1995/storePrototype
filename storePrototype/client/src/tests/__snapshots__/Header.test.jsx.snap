// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly Page should match snapshot 1`] = `ShallowWrapper {}`;
